sidecar/register/registration.go

```diff
@@ -82,12 +83,14 @@ func (agent *RegistrationAgent) Stop() {
 
 func (agent *RegistrationAgent) register() {
 	for {
-		registeredInstance, err := agent.config.Client.Register(agent.config.ServiceInstance)
-		if err == nil {
-			go agent.renew(registeredInstance)
-			return
+		if agent.checkAppHealth(DefaultReregistrationDelay) {
```

Comment:
Also had not considered starting and stopping the registration agent --that is a good idea.